test/simulator/timestamp/src/connection_simulator.cpp

```diff
@@ -52,13 +56,111 @@ connection_simulator::query_timestamp() const
     return (0);
 }
 
+/*
+ * Parse a single timestamp configuration string eg. oldest_timestamp=10. The string is split on the
+ * '=' and the new timestamp is assigned.
+ */
+int
+connection_simulator::parse_timestamp_config_single(
+  std::string config, uint64_t *new_oldest_ts, uint64_t *new_stable_ts)
+{
+
+    std::string ts_type;
+    uint64_t ts;
+
+    /* The substring before the '=' indicates the type of timestamp to be set. */
+    ts_type = config.substr(0, config.find("="));
+    /* Copy the substring after the '=' to get the timestamp value from the config string. */
+    std::string ts_string = config.substr(config.find("=") + 1);
+    /* Convert the timestamp to an int. */
+    ts = std::stoi(ts_string);
+
+    switch (system_timestamps_map[ts_type]) {
+    case oldest_timestamp:
+        *new_oldest_ts = ts;
+        break;
+    case stable_timestamp:
+        *new_stable_ts = ts;
+        break;
+    case durable_timestamp:
+        durable_ts = ts;
+        break;
+    }
+
+    return 0;
+}
+
+/*
+ *  Parse the timestamp configuration string with timestamps separated by ','.
+ */
 int
-connection_simulator::set_timestamp() const
+connection_simulator::parse_timestamp_config(
+  std::string config, uint64_t *new_oldest_ts, uint64_t *new_stable_ts)
 {
+    size_t pos = 0;
+    std::string token;
+
+    /* Loop over the timestamp configuration strings separated by ','. */
+    while ((pos = config.find(",")) != std::string::npos) {
+        token = config.substr(0, pos);
+        std::cout << token << std::endl;
+
+        parse_timestamp_config_single(token, new_oldest_ts, new_stable_ts);
+
+        config.erase(0, pos + 1);
+    }
+
+    /* Parse the final timestamp configuration string. */
+    parse_timestamp_config_single(config, new_oldest_ts, new_stable_ts);
+
+    return 0;
```

Comment:
Can the function return `void` instead of an `int`?

Reply:
Yeah, it doesn't need an `int`.